src/sys/DISK.ASM

```diff
@@ -29,19 +29,30 @@ verify:                                         ; Verify that a file exists
     mov si, iface_disk_search
     call comment                                ; Comment message
 
-    mov ax, 0x3800
-    mov es, ax                                  ; Segment the root directory is loaded in
-    mov di, 0x0000                              ; Start looking at offset 0
+    push 0x3800
+    pop es                                      ; Segment our directories are loaded in
+
+    mov cl, byte [dir_id]                       ; Each directory is of equal length
+    and cl, 0b00000001                          ; Max ID is 1
+    shl cx, 0xD                                 ; 0x2000 (8192 B) each
+    mov di, cx                                  ; This is the start of our directory
```

Comment:
This logic is added to shift through the filesystem memory area to the correct directory. The directory ID is 0 or 1 (to be extended), and corresponds to a specific 0x2000 area in memory.